core/audits/insights/image-delivery-insight.js

```diff
@@ -36,14 +34,39 @@ class ImageDeliveryInsight extends Audit {
    * @return {Promise<LH.Audit.Product>}
    */
   static async audit(artifacts, context) {
-    // TODO: implement.
     return adaptInsightToAuditProduct(artifacts, context, 'ImageDelivery', (insight) => {
+      if (!insight.optimizableImages.length) {
+        // TODO: show UIStrings.noOptimizableImages?
+        return;
+      }
+
+      const relatedEventsMap = insight.relatedEvents && !Array.isArray(insight.relatedEvents) ?
+        insight.relatedEvents :
+        null;
+
       /** @type {LH.Audit.Details.Table['headings']} */
       const headings = [
+        /* eslint-disable max-len */
+        {key: 'url', valueType: 'url', label: str_(i18n.UIStrings.columnURL), subItemsHeading: {key: 'reason', valueType: 'text'}},
+        {key: 'totalBytes', valueType: 'bytes', label: str_(i18n.UIStrings.columnResourceSize)},
+        {key: 'wastedBytes', valueType: 'bytes', label: str_(i18n.UIStrings.columnWastedBytes), subItemsHeading: {key: 'wastedBytes', valueType: 'bytes'}},
+        /* eslint-enable max-len */
       ];
+
       /** @type {LH.Audit.Details.Table['items']} */
-      const items = [
-      ];
+      const items = insight.optimizableImages.map(image => ({
+        url: image.request.args.data.url,
+        totalBytes: image.request.args.data.decodedBodyLength,
+        wastedBytes: image.byteSavings,
+        subItems: {
+          type: /** @type {const} */ ('subitems'),
+          items: (relatedEventsMap?.get(image.request) ?? []).map((reason, i) => ({
+            reason,
```

Comment:
This is a case where we should use a different UI string in Lighthouse. The UI string delivered here includes the byte savings but we don't need it because the byte savings can be put in a different table column.

<img width="723" alt="Screenshot 2025-02-10 at 9 03 35 AM" src="https://github.com/user-attachments/assets/fda82973-5645-45c9-a4e9-0c011f622aea" />

Reply:
I was initially thinking that we could have duplicate strings live in LH without the est savings tacked on, but then we wouldn't automatically get UI string changes.

Decomposing in RPP sounds good to me.